src/main/scala/xiangshan/frontend/IFU.scala

```diff
@@ -302,18 +300,28 @@ class NewIFU(implicit p: Parameters) extends XSModule
   .elsewhen(f0_fire && !f0_flush) {f1_valid  := true.B}
   .elsewhen(f1_fire)              {f1_valid  := false.B}
 
-  val f1_pc_adder_cut_point = (VAddrBits/2) - 1 // equal lower_result overflow bit
-  val f1_pc_high            = f1_ftq_req.startAddr(VAddrBits-1,f1_pc_adder_cut_point)
+  // val f1_pc_adder_cut_point = (VAddrBits/2) - 1 // equal lower_result overflow bit
+  // This is the point calculated for splicing PC,(VAddrBits/2) - 1 is too large,pc+32.U only need 6 bits
+  val pc_adder_cut_point = (VAddrBits/4) - 1 // equal lower_result overflow bit
+  val f1_pc_high            = f1_ftq_req.startAddr(VAddrBits-1,pc_adder_cut_point)
   val f1_pc_high_plus1      = f1_pc_high + 1.U
 
-  val f1_pc_lower_result    = VecInit((0 until PredictWidth).map(i => Cat(0.U(1.W), f1_ftq_req.startAddr(f1_pc_adder_cut_point-1, 0)) + (i * 2).U)) // cat with overflow bit
+  /**
+   * In order to reduce power consumption, avoid calculating the full PC value in the first level.
+   * code of original logic, this code has been deprecated
+   * val f1_pc                 = VecInit(f1_pc_lower_result.map{ i =>  
+   *  Mux(i(f1_pc_adder_cut_point), Cat(f1_pc_high_plus1,i(f1_pc_adder_cut_point-1,0)), Cat(f1_pc_high,i(f1_pc_adder_cut_point-1,0)))})
+   * 
+   */
+  val f1_pc_lower_result    = VecInit((0 until PredictWidth).map(i => Cat(0.U(1.W), f1_ftq_req.startAddr(pc_adder_cut_point-1, 0)) + (i * 2).U)) // cat with overflow bit
+  
   val f1_pc                 = VecInit(f1_pc_lower_result.map{ i =>  
-    Mux(i(f1_pc_adder_cut_point), Cat(f1_pc_high_plus1,i(f1_pc_adder_cut_point-1,0)), Cat(f1_pc_high,i(f1_pc_adder_cut_point-1,0)))})
+    Mux(i(pc_adder_cut_point), Cat(f1_pc_high_plus1,i(pc_adder_cut_point-1,0)), Cat(f1_pc_high,i(pc_adder_cut_point-1,0)))})
```

Comment:
2. This concatenate logic is repeated so many times that I would recommend encapsulate it. Something like:
```scala
  def CatPC(low: UInt, high: UInt, high1: UInt): UInt = {
      Mux(
        low(PcCutPoint),
        Cat(high1, low(PcCutPoint-1, 0)),
        Cat(high, low(PcCutPoint-1,0))
      )
  }
  def CatPC(lowVec: Vec[UInt], high: UInt, high1: UInt): Vec[UInt] = VecInit(lowVec.map(CatPC(_, high, high1)))
```
and then
```scala
val f1_pc = CatPC(f1_pc_lower_result, f1_pc_high, f1_pc_high_plus1)
```
etc.